### PR TITLE
Inclined to join flag fix

### PR DESCRIPTION
**Pull request by Kert**
Inlcined to join flag will always work except when there's no place in hero's army

### CODE REVIEW

File: src/cpp/shared/adventure/adv.cpp [hunk 1]
```diff
@@ -683,17 +685,52 @@ void advManager::QuickInfo(int x, int y) {
...
 	}
-	if (GetMapCellXY(cell, &x, &y)) {
+
+  int x, y;
+	if (GetMapCellXY(monsterCell, &x, &y)) {
 		ScriptCallback("OnMonsterInteract", x, y);
 	}
-	this->PlayerMonsterInteract_orig(cell, other, player, window, a1, a2, a3, a4, a5);
+
+  bool joinCellFlag = monsterCell->extraInfo & (1 << 12);
```
Comment (jkoppel) [inline src/cpp/shared/adventure/adv.cpp:699]: That 12 is a magic number and needs to be declared somewhere.

Where's the code for setting  this flag? It should already exist?

File: src/cpp/shared/adventure/adv.cpp [hunk 2]
```diff
@@ -683,17 +685,52 @@ void advManager::QuickInfo(int x, int y) {
...
+  int creatureIdx = monsterCell->objectIndex;
+  if(!joinCellFlag || !hro->army.CanJoin(creatureIdx) || gbInCampaign || (xIsPlayingExpansionCampaign && xCampaign.campaignID <= 3)) {
+    this->PlayerMonsterInteract_orig(monsterCell, anotherCell, hro, window, x1, y1, unused, x2, y2);
+    return;
+  }
+
+  gpMouseManager->ShowColorPointer();
+  int quantity = (unsigned char)monsterCell->extraInfo;
+  sprintf(gText, gEventText[66], GetCreaturePluralName(creatureIdx));
```
Comment (jkoppel) [inline src/cpp/shared/adventure/adv.cpp:708]: More magic numbers.

**Comment by jkoppel** [comment]
Where is the previous code for the inclinedToJoin flag? Did it just not exist?

**Comment by Kert** [comment]
> Where is the previous code for the inclinedToJoin flag? Did it just not exist?

It's all in src/cpp/shared/scripting/funcs.cpp that you can observe in Files changed. I just removed comments because they are obsolete now

File: src/cpp/shared/adventure/adv.cpp (hunk 2 above)
Comment (Kert) [reply src/cpp/shared/adventure/adv.cpp:708]: Would you like me to try and make an enum for all event texts?

**Comment by Kert** [comment]
Issues addressed anyway

**Comment by jkoppel** [comment]
Oh right! I remember now. We believed that setting this bit would make creatures inclined to join. However, the actual event logic was more complicated, and so it wouldn't always work.

**Comment by jkoppel** [comment]
Nice work!

**Pull request merged**